### PR TITLE
Improve: break in type declaration for variant and record

### PR DESCRIPTION
**Pull request by hhugo**
I could add an option to configure that but I'd only do it if some people prefer the old behavior.

### CODE REVIEW

**Comment by jberdine** [comment]
Yes, as an option please.

**Comment by hhugo** [comment]
option addded

File: src/Conf.ml [hunk 1]
```diff
@@ -163,6 +163,20 @@ let break_infix =
           default
       & info ["break-infix"] ~doc ~env)
 
+let type_decl =
+  let doc =
+    "Style of type declaration. Can be set in a config file with a \
+     `break-infix {wrap,fit-or-vertical}` line. `wrap` will group simple \
+     expressions and try to format them in a single line."
```
Comment (jberdine) [inline src/Conf.ml]: copypasta

File: src/Conf.ml [hunk 2]
```diff
@@ -163,6 +163,20 @@ let break_infix =
...
+  in
+  let env = Arg.env_var "OCAMLFORMAT_TYPE_DECL" in
+  let default = `Auto in
+  mk ~default
+    Arg.(
+      value
+      & opt (enum [("sparse", `Sparse); ("auto", `Auto)]) default
```
Comment (jberdine) [inline src/Conf.ml]: Nit: I think most cases order the alternatives with the default first.

File: src/Conf.ml [hunk 3]
```diff
@@ -358,6 +372,7 @@ type t =
   ; if_then_else: [`Compact | `Keyword_first]
   ; infix_precedence: [`Indent | `Parens]
   ; break_infix: [`Wrap | `Fit_or_vertical]
+  ; type_decl: [`Sparse | `Auto]
```
Comment (jberdine) [inline src/Conf.ml]: Same default-first nit.

**Comment by hhugo** [comment]
I've renamed `auto` into `compact` similar to `if-then-else`

**Pull request merged**